### PR TITLE
Add buy infrastructure to volume filter (part of #522)

### PR DESCRIPTION
**Pull request by debnil**
This PR adds the buy infrastructure to the volume filter, as per step 2 [here](https://github.com/stellar/kelp/pull/548#issuecomment-712792629).

### CODE REVIEW

File: plugins/volumeFilter.go
Comment (nikhilsaraf) [inline plugins/volumeFilter.go:226]: @debnil I think we are doing this PR too early. the next step is to fill in this function, and add the relevant tests to the buyVolume filter test (and any other tests related to the buy side).

once that's done, the refactor should be the next step. this will not be an exact cherry-pick from the previous set of commits I had added because of some of the volume filter changes we made.

as a last step we should add a new buy twap strategy with the config file. doing it in steps will make for a smaller PR.

**Review by nikhilsaraf — COMMENTED** [review]
as commented inline, I think we should hold off on this PR

File: plugins/volumeFilter_test.go
Comment (nikhilsaraf) [inline plugins/volumeFilter_test.go:131]: this should be `action` instead of hardcoding to `DailyVolumeActionSell`

File: queries/dailyVolumeByDate_test.go
Comment (nikhilsaraf) [inline queries/dailyVolumeByDate_test.go:51]: please add this case in a subsequent PR (better to separate out so we're not complicating this PR)

File: queries/dailyVolumeByDate_test.go
Comment (nikhilsaraf) [inline queries/dailyVolumeByDate_test.go:119]: was this a direct copy paste or did something change inside this too?

**Review by nikhilsaraf — COMMENTED** [review]
few minor comments, please fix those up and we should be good to merge.
can you leave the comments unresolved so I can check each item that is fixed up? will be a bit easier for me to track.

any leftover items, please address in a new PR

File: queries/dailyVolumeByDate_test.go
Comment (debnil) [reply queries/dailyVolumeByDate_test.go:51]: That was the plan - the TODO was so I'd remember this in the next step of implementation. Should I not do that?

File: queries/dailyVolumeByDate_test.go
Comment (debnil) [reply queries/dailyVolumeByDate_test.go:119]: Direct copy paste - nothing changed, just a small performance fix. 

Previously, the tests were setting up a brand new database in each iteration of the for-loop / for each test case. I found that this was making the run-time substantially slower. I moved the setup steps out of the loop and made it about 5x faster. (.124s vs .6s). I don't think this is happening on the CI, but it's a solid saving without losing any correctness.

File: plugins/volumeFilter.go
Comment (debnil) [reply plugins/volumeFilter.go:226]: We discussed this earlier today, but for documentation, this is the right stage for this fix. We can't implement this function absent the other infrastructural changes in this PR.

File: plugins/filterFactory_test.go
Comment (debnil) [reply plugins/filterFactory_test.go:164]: Sounds good. Since this is updating an old test, I figured that like our more recent tests, we'd want to vary both mode and action. To do this more efficiently, I changed the `configInput` to a format string and injected both mode and action. If the previous way would be preferred, let me know and I'm happy to change it back.

Comment (nikhilsaraf) [reply plugins/filterFactory_test.go:162]: the previous way is a little better because it's more readable, a direct mapping from input to output.

File: plugins/volumeFilter_test.go
Comment (nikhilsaraf) [inline plugins/volumeFilter_test.go:293]: these should be called `baseCapInBase` and `baseCapInQuote` instead of `sellBaseCapInBase` and `sellBaseCapInQuote`. Better to do this in a separate PR since the overhead of this PR is already getting complex. It's ok to do this when you add the `buy` action to this for loop.

File: plugins/volumeFilter_test.go
Comment (nikhilsaraf) [inline plugins/volumeFilter_test.go:290]: PR submission tip:
it's best not to add changes like this in follow-on PR changes since it introduces a large amount of red and green lines. As a reviewer, this makes reviewing this PR 10x more complex, since I have to ensure nothing has changed in here and whatever has changed makes sense, etc.
I say this now, because I'm hoping to start leaving similar tips on PRs that will allow us to increase our PR efficiency (such as the comment about not resolving review comments). Hopefully this will reduce the PR fatigue for both of us, and happy to get similar thoughts and tips from you as well :)

File: queries/dailyVolumeByDate_test.go
Comment (nikhilsaraf) [reply queries/dailyVolumeByDate_test.go:110]: yes, good fix! 👍

File: plugins/filterFactory_test.go
Comment (nikhilsaraf) [inline plugins/filterFactory_test.go:233]: @debnil add a check on the `action` field here

**Review by nikhilsaraf — COMMENTED** [review]
I think there's only 1 change below and we should be good to merge after that.
Any follow-on items can be done in a subsequent PR. thanks! 👍

**Review by nikhilsaraf — APPROVED** [review]
lgtm, thanks 🚢

**Pull request merged**
